src/QAT/qat/purr/integrations/tket.py

```diff
@@ -407,8 +407,11 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
             optimizations_failed = not optimize_circuit(circ, architecture, opts)
         else:
             coupling_qualities = list({val.quality for val in couplings})
-            coupling_qualities.sort(reverse=True)
-            for quality_level in coupling_qualities:
+            min_quality = min(coupling_qualities)
```

Comment:
For some live hardware this does not select the best qubits successfully. We need to hold off on merging this until that is resolved.

Reply:
There's also a trade-off between 'perfect' and 'usable' if one takes 50 years to process. There's next to no difference between the two algorithms except this one just batches numbers together.

I'd like to see some examples of where it dosen't pick good-enough qubits and the quality settings have been set correctly in the model.